jdrfandroidbleparser/src/main/java/org/ehealthinnovation/jdrfandroidbleparser/bgm/characteristic/common/Composable.kt

```diff
@@ -0,0 +1,21 @@
+package org.ehealthinnovation.jdrfandroidbleparser.bgm.characteristic.common
+
+import android.bluetooth.BluetoothGattCharacteristic
+
+/**
+ * Many base Bluetooth Characteristics not only have the requirement that they must decode a given
+ * [BluetoothGattCharacteristic] to primitive data types, they must also be able to do the reverse
+ * as well. These characteristics, must implement the [Composable] interface.
+ * [BaseCharacteristic] classes can just extend this class and not worry about implementation.
+ *
+ * Created by miantorno on 2018-06-19.
+ */
+interface Composable {
+
+    /**
+     * Compose a given characteristic into the resulting [BluetoothGattCharacteristic].
+     */
+    @Throws(IllegalStateException::class)
+    fun composeCharacteristic(): BluetoothGattCharacteristic
```

Comment:
We get to choose if we want to return a byte array or a BluetoothGattCharacteristic object. The workflow for writing a characteristic is getting an instance of  BluetoothGattCharacteristic from the android.bluetooth layer, then set it with a desired byte array. If we want to return a BluetoothGattCharacteristic, consider adding an input argument.

```
fun composeCharacteristic(c :BluetoothGattCharacteristic?):BluetoothGattCharacteristic
{
    val toReturn = c ?: BluetoothGattCharacteristic(dummy arguments)
    //composing
    toReturn.setValue(resultingArray)
    return c
}
```

Reply:
Why do we want to give people access to the raw byte data in the API we provide? Who will be using it and what purpose would it serve?